### PR TITLE
Break up ugly mixer-config.md and fix other gen issues

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR addresses a number of issues with the generated Mixer documentation:
- Fix Enum names
- Fix ordering of Service doc (should come before Status RPC doc)
- Removes mixer-config.md (and all of the problematic content that should not be exposed)
- Adds a policy-and-telemetry-rules.md doc to capture rule writing config in one spot
- Adds a value-type.md doc to put `ValueType` reference in one place
- Removes value type stuff from template docs.

### CODE REVIEW

File: _docs/reference/config/mixer/policy-and-telemetry-rules.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  handler: handler.prometheus
+  instances:
+  - RequestCount.metric
+```
+
+<a name="istio.mixer.v1.config.Handler"></a>
+### Handler
+Handler allows the operator to configure a specific adapter implementation.
+
+In the following example we define a `metric` handler using the Mixer's
```
Comment (guptasu) [inline _docs/reference/config/mixer/policy-and-telemetry-rules.md]: `prometheus` handler using the Mixer's

File: _docs/reference/config/mixer/policy-and-telemetry-rules.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+A Instance tells Mixer how to create values for particular template.
+
+Instance is defined by the operator. Instance is defined relative to a known
+template. Their purpose is to tell Mixer how to use attributes or literals to
+produce values for the specified template at runtime.
+
+The following example instructs Mixer to construct an instance associated with
+[metric template]({{home}}/docs/reference/config/mixer/template/metric.html). It
+provides a mapping from the template's fields to expressions. Instances produced
+with this instance can be referenced by [Actions](#istio.mixer.v1.config.Action)
```
Comment (guptasu) [inline _docs/reference/config/mixer/policy-and-telemetry-rules.md]: with this instance configuration

Comment (guptasu) [reply _docs/reference/config/mixer/policy-and-telemetry-rules.md]: can be referenced by [Actions](#istio.mixer.v1.config.Action) to map handlers to instances

File: _docs/reference/config/mixer/policy-and-telemetry-rules.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+ </tr>
+<a name="istio.mixer.v1.config.Action.instances"></a>
+ <tr>
+  <td><code>instances[]</code></td>
+  <td>repeated string</td>
+  <td>Required. Each value must match the fully qualified name of the <a href="#istio.mixer.v1.config.Instance">Instance</a>s. Referenced instances are evaluated by resolving the attributes/literals for all the fields. The constructed objects are then passed to the <code>handler</code> referenced within this action.</td>
+ </tr>
+</table>
+
+The following example instructs Mixer to invoke the `handler.prometheus` handler
```
Comment (guptasu) [inline _docs/reference/config/mixer/policy-and-telemetry-rules.md]: Consider matching this comment with the comment in the rule block above

Comment (douglas-reid) [reply _docs/reference/config/mixer/policy-and-telemetry-rules.md]: done.

File: _docs/reference/config/mixer/policy-and-telemetry-rules.md (hunk 1 above)
Comment (douglas-reid) [reply _docs/reference/config/mixer/policy-and-telemetry-rules.md]: done.

File: _docs/reference/config/mixer/policy-and-telemetry-rules.md (hunk 2 above)
Comment (douglas-reid) [reply _docs/reference/config/mixer/policy-and-telemetry-rules.md]: updated.

**Pull request merged**